File: src/main/java/ru/nedfreetoplay/querystorage/QueryStorage.java
package ru.nedfreetoplay.querystorage;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.client.model.ModelLoadingRegistry;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import ru.nedfreetoplay.querystorage.client.ClientQueryStorage;
import ru.nedfreetoplay.querystorage.client.ModelProvider;
import ru.nedfreetoplay.querystorage.util.ModBlocks;
import ru.nedfreetoplay.querystorage.util.ModBlockEntity;
import ru.nedfreetoplay.querystorage.util.ModItems;

public class QueryStorage implements ModInitializer {

	public static final String MOD_ID = "querystorage";
	public static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);

	@Override
	public void onInitialize() {

		ModItems.init();
		ModBlocks.init();
		ModBlockEntity.init();
		ModelLoadingRegistry.INSTANCE.registerResourceProvider(rm -> new ModelProvider());
	}
}

File: src/main/kotlin/ru/nedfreetoplay/querystorage/QueryStorage.kt
package ru.nedfreetoplay.querystorage

import com.mojang.logging.LogUtils
import net.minecraft.client.Minecraft
import net.minecraft.world.item.CreativeModeTab
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent
import net.minecraftforge.fml.event.lifecycle.FMLDedicatedServerSetupEvent
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries
import org.slf4j.Logger
import ru.nedfreetoplay.querystorage.registry.RegistryItem
import thedarkcolour.kotlinforforge.forge.MOD_BUS
import thedarkcolour.kotlinforforge.forge.runForDist

@Mod(QueryStorage.ID)
object QueryStorage {
    const val ID = "querystorage"

    // the logger for our mod
    val LOGGER: Logger = LogUtils.getLogger()
    val CreativeTab: CreativeModeTab = QueryCreativeTab(QueryStorage.ID)

    private val registryItems = DeferredRegister.create(ForgeRegistries.ITEMS, ID)

    init {
        LOGGER.info("Query Storage init!")

        // Register the KDeferredRegister to the mod-specific event bus
        //ModBlocks.REGISTRY.register(MOD_BUS)
        val items: Array<RegistryItem> = Items(
            registryItems,
            RegistryObjectList(
                RegistryItem()
            ))

        /*
        val items: Array<RegistryItem> = Items(registryItems
            RegistryObjectList(
                RegistryItem()
            ))
        */



        val obj = runForDist(
            clientTarget = {
                MOD_BUS.addListener(::onClientSetup)
                Minecraft.getInstance()
            },
            serverTarget = {
                MOD_BUS.addListener(::onServerSetup)
                "test"
            })

        println(obj)
    }

    /**
     * This is used for initializing client specific
     * things such as renderers and keymaps
     * Fired on the mod specific event bus.
     */
    private fun onClientSetup(event: FMLClientSetupEvent) {
    }

    /**
     * Fired on the global Forge bus.
     */
    private fun onServerSetup(event: FMLDedicatedServerSetupEvent) {
    }
}